src/librustc_typeck/check/_match.rs

```diff
@@ -690,7 +690,7 @@ https://doc.rust-lang.org/reference/types.html#trait-objects");
                     arm_span: arm.body.span,
                     source: match_src
                 });
-                coercion.coerce(self, &cause, &arm.body, arm_ty, self.diverges.get());
+                coercion.coerce(self, &cause, &arm.body, arm_ty);
```

Comment:
👍 -- the fact that this parameter was still needed should have been a warning sign. sigh.